File: C/insertion.c
#include "stdlib.h"
#include "stdio.h"

void insertion(int* tableau, int taille) {
    int i, j, tmp;
    for (i = 1; i < taille; i++) {
        tmp = tableau[i];
        j = i - 1;
        while (j >= 0 && tableau[j] > tmp) {
            tableau[j + 1] = tableau[j];
            j--;
        }
        tableau[j + 1] = tmp;
    }
}

int main() {

    int tableau[] = {5, 9, 3, 2, 1};

    insertion(tableau, 5);

    printf("%d %d %d %d %d\n", tableau[0], tableau[1], tableau[2], tableau[3], tableau[4]);

    return EXIT_SUCCESS;

}

File: Python/insertion.py
def insertion(liste):
    for i in range(1, len(liste)):
        j = i
        while j > 0 and liste[j] < liste[j - 1]:
            liste[j], liste[j - 1] = liste[j - 1], liste[j]
            j -= 1
    return liste

liste = [7, 2, 4, 1, 3, 5, 6]

liste = insertion(liste)

print(liste)